### ISSUE TITLE
How to add response headers to response?

### ISSUE THREAD

**Issue by flyaruu** [sammhicks/picoserve#3]
Hi, I'd like to return a response with a response header aside from just a body.
Before I just returned a &str, that worked fine, now I'm attempting this:

```rust
async fn get_root()-> impl IntoResponse {
    Response::new(OK, "hello world!")
        .with_headers(("Connection","close"))
    
}
```
But that errors:
```
type `picoserve::response::BodyHeaders` is private
```

Is this the correct way to do this?

**Comment by sammhicks** [comment]
Oops, that's a bug in my code!

I've now fixed it and pushed version 0.2.2 to crates.io, and improved the documentation.

As for your question, you can return headers in the following ways. For the full list, see the documentation for [response](https://docs.rs/picoserve/latest/picoserve/response/index.html) and [IntoResponse](https://docs.rs/picoserve/latest/picoserve/response/trait.IntoResponse.html)

```rust
async fn get_root()-> impl IntoResponse {
    (("Connection","close"), "hello world!")
}
``` 

```rust
async fn get_root()-> impl IntoResponse {
    (OK, ("Connection","close"), "hello world!")
}
``` 

```rust
async fn get_root()-> impl IntoResponse {
    Response::ok("hello world!")
        .with_header("Connection","close")
    
}
``` 

```rust
async fn get_root()-> impl IntoResponse {
    Response::new(OK, "hello world!")
        .with_header("Connection","close")
    
}
```

Thanks for the testing and feedback, it's very much appreciated :D !

**Comment by flyaruu** [comment]
Works great now!
